eng/pipelines/templates/stages/archetype-rust-release.yml

```diff
@@ -84,33 +85,77 @@ stages:
-                    $missingOwners = $additionalOwners | Where-Object { $existingOwners -notcontains $_ }
-
-                    foreach ($owner in $missingOwners) {
-                      Write-Host "> cargo owner --add $owner $crateName"
-                      cargo owner --add $owner $crateName
-                    }
-                  displayName: Publish Crate
-                  env:
-                    CARGO_REGISTRY_TOKEN: $(azure-sdk-cratesio-token)
+                  - pwsh: |
+                      Write-Host "##vso[task.setvariable variable=ArtifactIndex]0"
+                    displayName: Set ArtifactIndex to 0
+
+                  - ${{ each artifact in parameters.Artifacts }}:
+                    - pwsh: |
+                        # Read artifact release order from release-order.json 
+                        # and use ArtifactIndex to select the right one
+                        $index = [int]'$(ArtifactIndex)'
+                        $artifacts = Get-Content '$(Pipeline.Workspace)/drop/release-order.json' | ConvertFrom-Json
+                        if ($index -ge $artifacts.Count) {
+                          Write-Error "ArtifactIndex $index is out of range (0..$($artifacts.Count - 1))"
+                          exit 1
+                        }
+
+                        $artifactName = $artifacts[$index]
+                        Write-Host "Releasing artifact $artifactName (index $index)"
+
+                        $artifactRootPath = '$(Pipeline.Workspace)/drop'
+                        $outDir = '$(Pipeline.Workspace)/esrp-release'
+
+                        if (Test-Path $outDir) {
+                          Write-Host "Cleaning output directory: $outDir"
+                          Remove-Item -Path $outDir -Recurse -Force
+                        }
+                        New-Item -ItemType Directory -Path $outDir -Force | Out-Null
+
+                        Write-Host "Artifact name: $artifactName"
+
+                        $packageMetadataPath = "$artifactRootPath/PackageInfo/$artifactName.json"
+                        if (!(Test-Path $packageMetadataPath)) {
+                          Write-Error "Package metadata file not found: $packageMetadataPath"
+                          exit 1
+                        }
+
+                        $packageMetadata = Get-Content -Raw $packageMetadataPath | ConvertFrom-Json
+                        $packageVersion = $packageMetadata.version
+                        Write-Host "Package version: $packageVersion"
+
+                        $cratePath = "$artifactRootPath/$artifactName/$artifactName-$packageVersion.crate"
+                        Copy-Item `
+                          -Path $cratePath `
+                          -Destination $outDir
+                        Write-Host "Contents of $outDir"
+                        Get-ChildItem -Path $outDir | ForEach-Object { Write-Host $_.FullName }
+                      displayName: 'Copy crate for ESRP'
```

Comment:
There's a lot going on here we should put into `eng/scripts` instead so we can more easily test it without going through the painful commit->push->back to commit loop endemic of pipelines. Ideally, a script that takes simple inputs and spits out an order would be great.

Reply:
Add a shebang and make it executable. As with most rustaceans (same for gophers), most development is done on linux/WSL.